modules/simple_forms_api/spec/services/pdf_stamper_spec.rb

```diff
@@ -4,25 +4,33 @@
 require SimpleFormsApi::Engine.root.join('spec', 'spec_helper.rb')
 
 describe SimpleFormsApi::PdfStamper do
-  def self.test_pdf_stamp_error(stamp_method, test_payload)
-    it 'raises an error when generating stamped file' do
+  let(:data) { JSON.parse(File.read("modules/simple_forms_api/spec/fixtures/form_json/#{test_payload}.json")) }
+  let(:form) { "SimpleFormsApi::#{test_payload.titleize.gsub(' ', '')}".constantize.new(data) }
+
+  describe 'pdf_stamping' do
+    subject(:stamp) { SimpleFormsApi::PdfStamper.send(stamp_method, generated_form_path, form) }
+
+    before do
       allow(Common::FileHelpers).to receive(:random_file_path).and_return('fake/stamp_path')
       allow(Common::FileHelpers).to receive(:delete_file_if_exists)
-      allow(Prawn::Document).to receive(:generate).and_raise('Error generating stamped file')
-
-      generated_form_path = 'fake/generated_form_path'
-      data = JSON.parse(File.read("modules/simple_forms_api/spec/fixtures/form_json/#{test_payload}.json"))
-      form = "SimpleFormsApi::#{test_payload.titleize.gsub(' ', '')}".constantize.new(data)
+    end
 
-      expect do
-        SimpleFormsApi::PdfStamper.send(stamp_method, generated_form_path, form)
-      end.to raise_error(RuntimeError, 'Error generating stamped file')
+    context 'when generating stamped file' do
+      [
+        %w[stamp214142 vba_21_4142],
+        %w[stamp2110210 vba_21_10210],
+        %w[stamp21p0847 vba_21p_0847]
+      ].each do |stamp_method, test_payload|
+        context "when #{stamp_method} receives #{test_payload} payload" do
+          let(:test_payload) { test_payload }
+          let(:stamp_method) { stamp_method }
+          let(:generated_form_path) { 'fake/generated_form_path' }
 
-      expect(Common::FileHelpers).to have_received(:delete_file_if_exists).with('fake/stamp_path')
+          it 'raises an error' do
+            expect { stamp }.to raise_error(StandardError, 'An error occurred while verifying stamp.')
+          end
+        end
+      end
```

Comment:
I'd be curious to hear your thoughts on this refactor @Thrillberg. This wasn't necessary but I felt like it was a better structure to add new tests to. I'm happy to revert if you think it's unnecessary!